### PR TITLE
Redone error exit system

### LINKED ISSUE #23: Don't panic if the current directory is not a cargo project
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: CargoMetadata { stderr: "error: could not find `Cargo.toml` in `/home/joshua` or any parent directory\n" }', /home/joshua/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/cargo-supply-chain-0.0.2/src/common.rs:28:10
```

### PR DESCRIPTION
**Pull request by Owez**
Closes #23

### CODE REVIEW

**Comment by Shnatsel** [comment]
Thanks! I'll try to take a look later today. If I don't, please ping me.

File: src/common.rs [hunk 1]
```diff
@@ -13,11 +15,17 @@ pub struct SourcedPackage {
...
         .features(AllFeatures)
         .other_options(extra_options)
         .exec()
-        .unwrap();
+    {
+        Ok(v) => v,
+        Err(cargo_metadata::Error::CargoMetadata { stderr: _ }) => err_exit(
+            "Could not find Cargo.toml in current or parent directory, make sure your inside of a crate!",
+        ),
+        Err(_) => err_exit("Unknown error whilst fetching crate's metadata!"),
```
Comment (Shnatsel) [inline src/common.rs:27]: Why are we using custom error messages here? `MetadataCommand::new().exec()` already provides perfectly good error messages, I think we can just print them without any matching. I don't want to destroy the informative error messages we get from `cargo metadata` and show only "unknown error" which is not actionable for the user.

For reference, this is the output I get without this PR:

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: CargoMetadata { stderr: "error: could not find `Cargo.toml` in `/home/shnatsel` or any parent directory\n" }', /home/shnatsel/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-supply-chain-0.1.2/src/common.rs:20:10

**Review by Shnatsel — CHANGES_REQUESTED** [review]

**Comment by Shnatsel** [comment]
In the long term we probably want to return an error from this function instead of exiting the process, but that's not strictly necessary in this PR.

**Review by Shnatsel — APPROVED** [review]

**Comment by Shnatsel** [comment]
Ugh, I must have botched something in the conflict resolution, and there is no track of the manual changes made during resolution... thank you for nothing, git.

**Comment by Shnatsel** [comment]
Merged. Thanks!

**Comment by Owez** [comment]
When does git not mess up :)

**Pull request merged**